scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/internal/propertyAccessors.scala

```diff
@@ -1,37 +1,95 @@
 package pl.touk.nussknacker.engine.spel.internal
 
-import java.lang.reflect.{Method, Modifier}
-import java.util.Optional
-import org.apache.commons.lang3.ClassUtils
 import org.springframework.expression.spel.support.ReflectivePropertyAccessor
 import org.springframework.expression.{EvaluationContext, PropertyAccessor, TypedValue}
+import org.springframework.util.ClassUtils
 import pl.touk.nussknacker.engine.api.dict.DictInstance
 import pl.touk.nussknacker.engine.api.exception.NonTransientException
+import pl.touk.nussknacker.engine.definition.clazz.ClassDefinitionSet
 
+import java.lang.reflect.{Field, Method, Modifier}
+import java.util.Optional
 import scala.collection.concurrent.TrieMap
 
 object propertyAccessors {
 
   // Order of accessors matters - property from first accessor that returns `true` from `canRead` will be chosen.
   // This general order can be overridden - each accessor can define target classes for which it will have precedence -
   // through the `getSpecificTargetClasses` method.
-  def configured(): Seq[PropertyAccessor] = {
+  def configured(
+      classDefinitionSet: ClassDefinitionSet,
+      checkAccess: Boolean
+  ): Seq[PropertyAccessor] = {
+
     Seq(
       MapPropertyAccessor, // must be before NoParamMethodPropertyAccessor and ReflectivePropertyAccessor
-      new ReflectivePropertyAccessor(),
-      NullPropertyAccessor,              // must be before other non-standard ones
-      ScalaOptionOrNullPropertyAccessor, // must be before scalaPropertyAccessor
-      JavaOptionalOrNullPropertyAccessor,
-      PrimitiveOrWrappersPropertyAccessor,
-      StaticPropertyAccessor,
+      new CheckedReflectivePropertyAccessor(classDefinitionSet, checkAccess),
+      NullPropertyAccessor, // must be before other non-standard ones
+      new ScalaOptionOrNullPropertyAccessor(
+        classDefinitionSet,
+        checkAccess
+      ), // must be before scalaPropertyAccessor
+      new JavaOptionalOrNullPropertyAccessor(classDefinitionSet, checkAccess),
+      new PrimitiveOrWrappersPropertyAccessor(classDefinitionSet, checkAccess),
+      new StaticPropertyAccessor(classDefinitionSet, checkAccess),
       TypedDictInstancePropertyAccessor, // must be before NoParamMethodPropertyAccessor
-      NoParamMethodPropertyAccessor,
+      new NoParamMethodPropertyAccessor(classDefinitionSet, checkAccess),
       // it can add performance overhead so it will be better to keep it on the bottom
       MapLikePropertyAccessor,
       MapMissingPropertyToNullAccessor, // must be after NoParamMethodPropertyAccessor
     )
   }
 
+  class CheckedReflectivePropertyAccessor(
```

Comment:
It is just a theoretical question, I want to know the limitation of this method checking. Let's assume that we have some variable that could have multiple types of values. Am I right that once when canRead pass method call, we won't check if method call is allowed for other types?